src/EnergyPlus/PluginManager.cc

```diff
@@ -108,6 +108,13 @@ void registerNewCallback(EnergyPlusData &state, EMSManager::EMSCallFrom iCalledF
     state.dataPluginManager->callbacks[iCalledFrom].push_back(f);
 }
 
+void registerUserDefinedCallback(EnergyPlusData &state, const std::function<void(void *)> &f, const std::string &programNameInInputFile)
+{
+    // internally, E+ will UPPER the program name; we should upper the passed in registration name so it matches
+    state.dataPluginManager->userDefinedCallbackNames.push_back(Util::makeUPPER(programNameInInputFile));
+    state.dataPluginManager->userDefinedCallbacks.push_back(f);
+}
```

Comment:
This is the internal E+ function which registers the user-defined callback, pairing it with the matching passed in program name.